apps/crn-server/src/data-providers/users.ts

```diff
@@ -0,0 +1,338 @@
+import { GenericError } from '@asap-hub/errors';
+import { UserDataObject, UserPatchDataObject } from '@asap-hub/model';
+import {
+  config,
+  RestUser,
+  sanitiseForSquidex,
+  SquidexGraphqlClient,
+  SquidexRest,
+  SquidexRestClient,
+} from '@asap-hub/squidex';
+import Intercept from 'apr-intercept';
+import FormData from 'form-data';
+import { Got } from 'got';
+import mime from 'mime-types';
+import {
+  FetchUserQuery,
+  FetchUserQueryVariables,
+  FetchUsersQuery,
+  FetchUsersQueryVariables,
+} from '../autogenerated-gql/graphql';
+import { FetchUsersOptions } from '../controllers/users';
+import {
+  parseGraphQLUserToDataObject,
+  parseUserToDataObject,
+} from '../entities';
+import { FETCH_USER, FETCH_USERS } from '../queries/users.queries';
+import { fetchOrcidProfile, transformOrcidWorks } from '../utils/fetch-orcid';
+
+export interface UserDataProvider {
+  fetchById(id: string): Promise<UserDataObject | null>;
+  update(id: string, update: UserPatchDataObject): Promise<void>;
+  fetch(options: FetchUsersOptions): Promise<UserDataObject[]>;
+  fetchByCode(code: string): Promise<UserDataObject[]>;
+  updateAvatar(id: string, avatar: Buffer, contentType: string): Promise<void>;
+  connectByCode(
+    welcomeCode: string,
+    userId: string,
+  ): Promise<UserDataObject | null>;
+  syncOrcidProfile(
+    id: string,
+    cachedUser?: RestUser | undefined,
+  ): Promise<UserDataObject>;
+}
+export default function createUserDataProvider(
```

Comment:
I feel like since we use classes for the controllers we should continue to use them for data providers. I don't mind switching to factories but I would prefer to be consistent